datahub-web-react/src/app/govern/structuredProperties/StructuredProperties.tsx

```diff
@@ -11,54 +11,118 @@ import {
+import { useGetAutoCompleteResultsLazyQuery, useGetSearchResultsForMultipleQuery } from '@src/graphql/search.generated';
+import { Entity, EntityType, SortOrder, StructuredPropertyEntity } from '@src/types.generated';
+
+const MAX_PROPERTIES_TO_FETCH = 20;
 
 const StructuredProperties = () => {
     const isShowNavBarRedesign = useShowNavBarRedesign();
     const [searchQuery, setSearchQuery] = useState<string>('');
+    const [searchResults, setSearchResults] = useState<Entity[] | null>(null);
+    const [newProperty, setNewProperty] = useState<StructuredPropertyEntity>();
+    const [updatedProperty, setUpdatedProperty] = useState<StructuredPropertyEntity>();
+    const [totalCount, setTotalCount] = useState<number>(0);
+
     const [isDrawerOpen, setIsDrawerOpen] = useState<boolean>(false);
     const [isViewDrawerOpen, setIsViewDrawerOpen] = useState<boolean>(false);
-    const [selectedProperty, setSelectedProperty] = useState<SearchResult | undefined>();
+    const [selectedProperty, setSelectedProperty] = useState<StructuredPropertyEntity | undefined>();
     const me = useUserContext();
     const canEditProps = me.platformPrivileges?.manageStructuredProperties;
 
-    const handleSearch = (value) => {
-        setSearchQuery(value);
-    };
+    const getInputVariables = useCallback(
+        (start: number, count: number) => ({
+            types: [EntityType.StructuredProperty],
+            query: searchQuery,
+            start,
+            count,
+            searchFlags: { skipCache: true },
+            sortInput: {
+                sortCriteria: [{ field: CREATED_TIME_FIELD_NAME, sortOrder: SortOrder.Descending }],
+            },
+        }),
+        [searchQuery],
+    );
 
-    const inputs = {
-        types: [EntityType.StructuredProperty],
-        query: '',
-        start: 0,
-        count: 500,
-        searchFlags: { skipCache: true },
-    };
+    const inputs = useMemo(() => getInputVariables(0, MAX_PROPERTIES_TO_FETCH), [getInputVariables]);
 
-    // Execute search
     const { data, loading, refetch } = useGetSearchResultsForMultipleQuery({
-        variables: {
-            input: inputs,
-        },
+        variables: { input: inputs },
+        skip: !!searchQuery,
         fetchPolicy: 'cache-first',
     });
 
+    const [getAutoComplete, { data: autocompleteData }] = useGetAutoCompleteResultsLazyQuery();
+
+    const handleSearch = (value) => {
+        setSearchQuery(value);
+    };
+
+    useEffect(() => {
+        if (!searchQuery) {
+            setSearchResults(null);
+            return;
+        }
+
+        getAutoComplete({
+            variables: {
+                input: {
+                    query: searchQuery,
+                    field: DISPLAY_NAME_FILTER_NAME,
+                    limit: 200,
+                    type: EntityType.StructuredProperty,
+                },
+            },
+        });
+    }, [searchQuery, getAutoComplete]);
```

Comment:
instead of a `useEffect` could we make this a not lazy query, but use a reqular query which should update every time `searchQuery` changes? and then `skip` if there is no search query?

or even better, if we do a lazy query can we make it debounced so we don't make a call with every key stroke? i know we have examples of this throughout the app